src/main/java/net/snowflake/ingest/connection/SecurityManager.java

```diff
@@ -110,14 +120,26 @@
    * regenerateToken - Regenerates our Token given our current user,
    *                    account and keypair
    */
-  private void regenerateToken()
+  public void regenerateToken()
   {
     //create our JWT claim object
     JwtClaims claims = new JwtClaims();
 
     //set the issuer to the fully qualified username
-    claims.setIssuer(account + "." + user);
-    LOGGER.info("Creating Token with Issuer {}", account + "." + user);
+    String publicKeyFPInJwt = null;
+    try
+    {
+      publicKeyFPInJwt = getPublicKeyFp(savePublicKey(keyPair.getPublic()));
+      publicKeyFingerPrint = publicKeyFPInJwt;
+    }
+    catch(Exception e)
+    {
+      LOGGER.info("Failed to Generate public key fingerprint");
+    }
+
+    //set the subject to the fully qualified username
+    claims.setSubject(account + "." + user);
```

Comment:
I am a little confused here. Where exactly are we adding the publicKeyFP to the token? We don't seem to be doing that.